crmsh/bootstrap.py

```diff
@@ -2466,61 +2400,69 @@ def bootstrap_remove(context):
+
+    _context.cluster_node = get_cluster_node_hostname()
 
-    if not force and not confirm("Removing node \"{}\" from the cluster: Are you sure?".format(cluster_node)):
+    if not force_flag and not confirm("Removing node \"{}\" from the cluster: Are you sure?".format(_context.cluster_node)):
         return
 
-    if remove_localhost_check():
-        if not config.core.force and not force:
+    if _context.cluster_node == utils.this_node():
+        if not force_flag:
             error("Removing self requires --force")
-        # get list of cluster nodes
-        me = utils.this_node()
-        nodes = xmlutil.listnodes(include_remote_nodes=False)
-        othernode = next((x for x in nodes if x != me), None)
-        if othernode is not None:
-            # remove from other node
-            cmd = "crm cluster remove{} -c {}".format(" -y" if yes_to_all else "", me)
-            rc = utils.ext_cmd_nosudo("ssh{} -o StrictHostKeyChecking=no {} '{}'".format("" if yes_to_all else " -t", othernode, cmd))
-            if rc != 0:
-                error("Failed to remove this node from {}".format(othernode))
-        else:
-            # stop cluster
-            if not stop_service("corosync"):
-                error("Stopping corosync failed")
-
-            # remove all trace of cluster from this node
-            # delete configuration files from the node to be removed
-            toremove = [
-                SYSCONFIG_SBD,
-                CSYNC2_CFG,
-                corosync.conf(),
-                CSYNC2_KEY,
-                COROSYNC_AUTH,
-                "/var/lib/heartbeat/crm/*",
-                "/var/lib/pacemaker/cib/*"]
-            if not invoke('bash -c "rm -f {}"'.format(" ".join(toremove))):
-                error("Deleting the configuration files failed")
-    else:
+        remove_self()
+        return
+
+    if _context.cluster_node in xmlutil.listnodes():
         remove_node_from_cluster()
+    else:
+        error("Specified node {} is not configured in cluster! Unable to remove.".format(_context.cluster_node))
+
+
+def remove_self():
+    me = _context.cluster_node
+    yes_to_all = _context.yes_to_all
+    nodes = xmlutil.listnodes(include_remote_nodes=False)
+    othernode = next((x for x in nodes if x != me), None)
+    if othernode is not None:
+        # remove from other node
+        cmd = "crm cluster remove{} -c {}".format(" -y" if yes_to_all else "", me)
+        rc = utils.ext_cmd_nosudo("ssh{} -o StrictHostKeyChecking=no {} '{}'".format("" if yes_to_all else " -t", othernode, cmd))
+        if rc != 0:
+            error("Failed to remove this node from {}".format(othernode))
+    else:
+        # stop cluster
+        if not stop_service("corosync"):
+            error("Stopping corosync failed")
+        # remove all trace of cluster from this node
+        # delete configuration files from the node to be removed
+        toremove = [SYSCONFIG_SBD, CSYNC2_CFG, corosync.conf(), CSYNC2_KEY,
```

Comment:
Maybe we could put these list as a constant. I have already seen it in other place.

Reply:
OK, changed:)